### PR TITLE
[installation] - rebuild assets table removing com_massmail

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #13551 .

- rebuild `#__assets`  in installation sql  for (mysql,postgresql,mssql)
- rebuild  `#__assets` in sample_testing sql  for (mysql,postgresql,mssql)
- rebuild  `#__assets` in sample_blog sql  for (mysql,postgresql,mssql)
- rebuild  `#__assets` in sample_data sql  for (mysql,postgresql,mssql)
- rebuild  `#__assets` insample_learn sql  for (mysql,postgresql,mssql)
- rebuild  `#__assets` in sample_brochure sql  for (mysql,mssql)

### Summary of Changes
remove old components from the  `#__assets`  table like com_massmail
rebuild `#__assets` for  sql installation file

### Test instructions
install a fresh joomla site with sample datasets all work as before

### CODE REVIEW

**Comment by sanderpotjer** [comment]
I have tested this item :red_circle: unsuccessfully on 7165668e3fd812b07426778dedc0e83e5010ce32<br /><br />Script is currently not working, missing a semicolon on line 1825 of administrator/components/com_admin/script.php<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13587">issues.joomla.org/tracker/joomla-cms/13587</a>.</sub>

**Comment by sanderpotjer** [comment]
I have tested this item :red_circle: unsuccessfully on 7165668e3fd812b07426778dedc0e83e5010ce32<br /><br />Script is currently not working, missing a semicolon on line 1825 of administrator/components/com_admin/script.php<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13587">issues.joomla.org/tracker/joomla-cms/13587</a>.</sub>

**Comment by alikon** [comment]
damned c&p ;)

**Comment by andrepereiradasilva** [comment]
Afaik You need to remove it instalattion sql too

**Comment by sanderpotjer** [comment]
There are more references to the com_massmail in the installation sql files, like the sample_blog, sample_brochure, sample_data, sample_learn and sample_testing. And of course for the sqlazure, mysql and postgresql versions. This should be removed too.

**Comment by alikon** [comment]
i know, i need to find a bit free time
consider a work in progress for now

File: administrator/components/com_admin/script.php [hunk 1]
```diff
@@ -1843,6 +1847,33 @@ public function updateAssets()
...
+				// Install failed, roll back changes
+				$this->parent->abort(JText::sprintf('JLIB_INSTALLER_ABORT_COMP_INSTALL_ROLLBACK', $asset->stderr(true)));
+
+				return false;
+			}
+		}
+
+		$asset = JTable::getInstance('Asset');
+
+		if (!$asset->rebuild())
```
Comment (mbabker) [inline administrator/components/com_admin/script.php]: I would be careful about putting this into the update routine in a spot that will be run on every update.  There's a possible performance implication with a large table or further corrupting an already broken table.

Comment (alikon) [inline administrator/components/com_admin/script.php]: as usual you are right

**Comment by alikon** [comment]
it's ready for test now

**Comment by marrouchi** [comment]
@alikon Can please provice clear test instructions ? I wasn't able to apply the patch on a fresh clone of the repo using `git apply --check 13587.patch`

**Comment by alikon** [comment]
1. install latest staging with one of the sample data provided or with no data (multilingual)
   2.a install com_patchtester and apply this patch  or 2.b (i'm not a git expert)
   2.b `git apply --check 13587.patch`
3. delete configuration.php
4. install again  with the same data choosed at 1.)
5. check that all work as before
6. check that in the `#__assets` table there are no occurrencies with com_massmail

repeat 1 thru 6  for all kind of sample data

**Comment by marrouchi** [comment]
I have tested this item :white_check_mark: successfully on 217edaa5414451b8d4cdd30f2161a8aa1db0c764<br /><br />Thank you for the test instructions !

Tested on mysql DB.

Note that database is clean but there still occurrencies in files.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13587">issues.joomla.org/tracker/joomla-cms/13587</a>.</sub>

**Comment by alikon** [comment]
can you specify  

>  there still occurrencies in files.

**Comment by marrouchi** [comment]
Sorry, ignore my last comment. They are occurencies in com_users that has a mass mail feature.

File: installation/sql/mysql/joomla.sql [hunk 2]
```diff
@@ -44,47 +44,44 @@ INSERT INTO `#__assets` (`id`, `parent_id`, `lft`, `rgt`, `level`, `name`, `titl
...
--- --------------------------------------------------------
+(14, 1, 31, 32, 1, 'com_media', 'com_media', '{"core.admin":{"7":1},"core.manage":{"6":1},"core.create":{"3":1},"core.delete":{"5":1}}'),
+(15, 1, 33, 36, 1, 'com_menus', 'com_menus', '{"core.admin":{"7":1}}'),
+(16, 1, 37, 38, 1, 'com_messages', 'com_messages', '{"core.admin":{"7":1},"core.manage":{"7":1}}'),
+(17, 1, 39, 70, 1, 'com_modules', 'com_modules', '{"core.admin":{"7":1}}'),
+(18, 1, 71, 74, 1, 'com_newsfeeds', 'com_newsfeeds', '{"core.admin":{"7":1},"core.manage":{"6":1}}'),
+(19, 1, 75, 76, 1, 'com_plugins', 'com_plugins', '{"core.admin":{"7":1}}'),
+(20, 1, 77, 78, 1, 'com_redirect', 'com_redirect', '{"core.admin":{"7":1}}'),
+(21, 1, 79, 80, 1, 'com_search', 'com_search', '{"core.admin":{"7":1},"core.manage":{"6":1}}'),
+(22, 1, 81, 83, 1, 'com_templates', 'com_templates', '{"core.admin":{"7":1}}'),
```
Comment (csthomas) [inline installation/sql/mysql/joomla.sql]: IMO you should fix a few numbers on this line and on a few mentioned below.
Replace from 83 to 82

File: installation/sql/postgresql/joomla.sql [hunk 3]
```diff
@@ -42,47 +42,46 @@ INSERT INTO "#__assets" ("id", "parent_id", "lft", "rgt", "level", "name", "titl
...
-SELECT setval('#__assets_id_seq', 55, false);
+(14, 1, 31, 32, 1, 'com_media', 'com_media', '{"core.admin":{"7":1},"core.manage":{"6":1},"core.create":{"3":1},"core.delete":{"5":1}}'),
+(15, 1, 33, 36, 1, 'com_menus', 'com_menus', '{"core.admin":{"7":1}}'),
+(16, 1, 37, 38, 1, 'com_messages', 'com_messages', '{"core.admin":{"7":1},"core.manage":{"7":1}}'),
+(17, 1, 39, 70, 1, 'com_modules', 'com_modules', '{"core.admin":{"7":1}}'),
+(18, 1, 71, 74, 1, 'com_newsfeeds', 'com_newsfeeds', '{"core.admin":{"7":1},"core.manage":{"6":1}}'),
+(19, 1, 75, 76, 1, 'com_plugins', 'com_plugins', '{"core.admin":{"7":1}}'),
+(20, 1, 77, 78, 1, 'com_redirect', 'com_redirect', '{"core.admin":{"7":1}}'),
+(21, 1, 79, 80, 1, 'com_search', 'com_search', '{"core.admin":{"7":1},"core.manage":{"6":1}}'),
+(22, 1, 81, 83, 1, 'com_templates', 'com_templates', '{"core.admin":{"7":1}}'),
```
Comment (csthomas) [inline installation/sql/postgresql/joomla.sql]: Replace from 83 to 82

File: installation/sql/sqlazure/joomla.sql [hunk 4]
```diff
@@ -60,83 +60,81 @@ SELECT 12, 1, 27, 28, 1, 'com_login', 'com_login', '{}'
 UNION ALL
 SELECT 13, 1, 29, 30, 1, 'com_mailto', 'com_mailto', '{}'
 UNION ALL
-SELECT 14, 1, 31, 32, 1, 'com_massmail', 'com_massmail', '{}'
+SELECT 14, 1, 31, 33, 1, 'com_media', 'com_media', '{"core.admin":{"7":1},"core.manage":{"6":1},"core.create":{"3":1},"core.delete":{"5":1}}'
```
Comment (csthomas) [inline installation/sql/sqlazure/joomla.sql]: Replace from 33 to 32

File: installation/sql/sqlazure/joomla.sql [hunk 5]
```diff
@@ -60,83 +60,81 @@ SELECT 12, 1, 27, 28, 1, 'com_login', 'com_login', '{}'
...
+SELECT 19, 1, 75, 76, 1, 'com_plugins', 'com_plugins', '{"core.admin":{"7":1}}'
 UNION ALL
-SELECT 20, 1, 77, 78, 1, 'com_plugins', 'com_plugins', '{"core.admin":{"7":1}}'
+SELECT 20, 1, 77, 78, 1, 'com_redirect', 'com_redirect', '{"core.admin":{"7":1}}'
 UNION ALL
-SELECT 21, 1, 79, 80, 1, 'com_redirect', 'com_redirect', '{"core.admin":{"7":1}}'
+SELECT 21, 1, 79, 80, 1, 'com_search', 'com_search', '{"core.admin":{"7":1},"core.manage":{"6":1}}'
 UNION ALL
-SELECT 22, 1, 81, 82, 1, 'com_search', 'com_search', '{"core.admin":{"7":1},"core.manage":{"6":1}}'
+SELECT 22, 1, 81, 83, 1, 'com_templates', 'com_templates', '{"core.admin":{"7":1}}'
```
Comment (csthomas) [inline installation/sql/sqlazure/joomla.sql]: Replace from 83 to 82

Comment (alikon) [inline installation/sql/sqlazure/joomla.sql]: fixed

File: installation/sql/sqlazure/joomla.sql (hunk 4 above)
Comment (alikon) [inline installation/sql/sqlazure/joomla.sql]: fixed

File: installation/sql/postgresql/joomla.sql (hunk 3 above)
Comment (alikon) [inline installation/sql/postgresql/joomla.sql]: fixed

File: installation/sql/mysql/joomla.sql (hunk 2 above)
Comment (alikon) [inline installation/sql/mysql/joomla.sql]: fixed

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on d8c76e56280eaf847d03b5cd4f02edc7bdc99830<br /><br />Mysql and postgresql<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13587">issues.joomla.org/tracker/joomla-cms/13587</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
at Point 3 got:
bildschirmfoto 2017-02-22 um 19 14 10 kopie

**Comment by franz-wohlkoenig** [comment]
I have tested this item :red_circle: unsuccessfully on d8c76e56280eaf847d03b5cd4f02edc7bdc99830<br /><br />- /index.php/single-news-feed – 404 Error. Feed not found
- /index.php/feed-display – Feed not found. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13587">issues.joomla.org/tracker/joomla-cms/13587</a>.</sub>

**Pull request closed without merging**